bpf/kmesh/workload/include/authz.h

```diff
@@ -182,13 +185,312 @@ static int match_dst_ports(Istio__Security__Match *match, struct xdp_info *info,
     return UNMATCHED;
 }
 
+static inline __u32 convert_ipv4_to_u32(const struct ProtobufCBinaryData *ipv4_data)
+{
+    if (!ipv4_data->data || ipv4_data->len != 4) {
+        return 0;
+    }
+
+    unsigned char *data = (unsigned char *)KMESH_GET_PTR_VAL(ipv4_data->data, unsigned char);
+    if (!data) {
+        BPF_LOG(INFO, AUTH, "convert_ipv4_to_u32: Failed to read data from ipv4_data\n");
+        return 0;
+    }
+
+    return (data[3] << 24) | (data[2] << 16) | (data[1] << 8) | (data[0] << 0);
+}
+
+static inline __u32 convert_ipv6_to_u32(struct ip_addr *rule_addr, const struct ProtobufCBinaryData *ipv6_data)
+{
+    if (!rule_addr || !ipv6_data)
+        return 1;
+    if (!ipv6_data->data || ipv6_data->len != 16) {
+        return 1;
+    }
+
+    unsigned char *v6addr = (unsigned char *)KMESH_GET_PTR_VAL(ipv6_data->data, unsigned char *);
+    if (!v6addr) {
+        return 1;
+    }
+
+    for (int i = 0; i < 4; i++) {
+        for (int j = 0; j < 4; j++) {
+            rule_addr->ip6[i] |= (v6addr[i * 4 + j] << (i * 8));
+        }
+    }
+
+    return 0;
+}
+
+// reference cilium https://github.com/cilium/cilium/blob/main/bpf/lib/ipv6.h#L122
+#define GET_PREFIX(PREFIX) bpf_htonl(PREFIX <= 0 ? 0 : PREFIX < 32 ? ((1 << PREFIX) - 1) << (32 - PREFIX) : 0xFFFFFFFF)
+
+static inline void ipv6_addr_clear_suffix(union v6addr *addr, int prefix)
+{
+    addr->p1 &= GET_PREFIX(prefix);
+    prefix -= 32;
+    addr->p2 &= GET_PREFIX(prefix);
+    prefix -= 32;
+    addr->p3 &= GET_PREFIX(prefix);
+    prefix -= 32;
+    addr->p4 &= GET_PREFIX(prefix);
+}
+
+static inline int matchIpv4(__u32 ruleIp, __u32 preFixLen, __be32 targetIP)
+{
+    __u32 mask = 0;
+
+    if (preFixLen > 32) {
+        return UNMATCHED;
+    }
+    mask = 0xFFFFFFFF << (32 - preFixLen);
+    if ((ruleIp & mask) == (targetIP & mask)) {
+        return MATCHED;
+    }
+    return 0;
+}
+
+static inline int matchIpv6(struct ip_addr *rule_addr, struct ip_addr *target_addr, __u32 prefixLen)
+{
+    if (prefixLen > 128)
+        return UNMATCHED;
+
+    ipv6_addr_clear_suffix(target_addr, prefixLen);
+    if (rule_addr->ip6[0] == target_addr->ip6[0] && rule_addr->ip6[1] == target_addr->ip6[1]
+        && rule_addr->ip6[2] == target_addr->ip6[2] && rule_addr->ip6[3] == target_addr->ip6[3]) {
+        BPF_LOG(DEBUG, KMESH, "match ipv6\n");
+        return MATCHED;
+    }
+
+    return UNMATCHED;
+}
+
+static inline int
+matchIp(struct ProtobufCBinaryData *addrInfo, __u32 preFixLen, struct bpf_sock_tuple *tuple_info, __u8 type)
+{
+    if (!addrInfo || addrInfo->len == 0) {
+        BPF_LOG(ERR, AUTH, "addrInfo is NULL or length is 0\n");
+        return UNMATCHED;
+    }
+
+    if (addrInfo->len == 4) {
+        __u32 rule_ip = convert_ipv4_to_u32(addrInfo);
+        if (type & TYPE_SRCIP) {
+            BPF_LOG(
```

Comment:
Is this necessary log?

Reply:
1.It seems more appropriate to merge the two logs into one.  Because the bpf program has a limit on the number of instructions；
2.It is recommended that you do a check for logs elsewhere.